datafusion/src/logical_plan/plan.rs

```diff
@@ -38,7 +38,7 @@ pub enum JoinType {
     Inner,
     /// Left Join
     Left,
-    /// Right Join
+    /// Right Join: will be transferred to Left Join
```

Comment:
This is a good idea @xudong963  -- I do think it will keep things somewhat easier to handle.

If we are going to make this change, I think we should remove the `Right` enum value from `JoinType` 

The only possibly issue I can see with this change is that is may complicate the join implementation (which needs to distinguish between both which relation needs to be preserved and which gets hashed)

Using  `Left` and `Right` enum values might allow us to maintain that distinction 

So for example

```
A LEFT JOIN B    --> Means preserve B, hash B
B LEFT JOIN A    --> Means preserve A, hash A
B RIGHT JOIN A --> Means preserve B, hash A
A RIGHT JOIN B --> Means preserve A, hash B
```


Thoughts @Dandandan ?

Reply:
Won't removing the right join effectively mean we can't reorder a left join to a right join?

When a left join has a bigger side on the left it means we would like to swap the sides and the join implementation.

Besides hashing there is also the producing rows that didn't match. For a left join this is done at the end, while for the right join this can be done for each `RecordBatch`.

It might be a bit strange to encode a different left/right enum, because for inner, outer, etc. that doesn't make a lot of sense?

Reply:
Yes I think you are right that if we removed the `Right` variant then we would have to add some other way to encode the "which to hash"

I find the standard terminology confusing because there is an (implicit) optimization for one of the two join inputs in most implementations (e.g. hash join, which one to hash) that is often encoded as "which appears as the left argument in the tree" which has a different meaning than a `LEFT JOIN`